src/powershell/Add-ExportScopeConfiguration.psm1

```diff
@@ -0,0 +1,137 @@
+        [Parameter()]
+        [String]
+        [ValidateNotNullOrEmpty()]
+        $TenantId
+        
+    )
+    $ErrorActionPreference = 'Stop'
+    [string]$operation = 'create'
+
+    # Main
+    Write-Output ''
+    Write-Output ("{0}    Starting" -f (Get-Date))
+
+    if (!$Scope.StartsWith('/')) {
+        $Scope = '/' + $Scope
+    }
+
+    if ($Scope.EndsWith('/')) {
+        $Scope = $Scope.Substring(0, $Scope.Length - 1)
+    }
+
+    Write-Output ("{0}    Export Scope to add: {1}" -f (Get-Date), $Scope)
+    Write-Output ("{0}    tenantId for scope: {1}" -f (Get-Date), $TenantId)
+
+    $resourceGroup = Get-AzResourceGroup -Name $HubName -ErrorAction SilentlyContinue
+    if ($null -eq $resourceGroup) {
+        Write-Output ("{0}    FinOps Hub {1} Not Found" -f (Get-Date), $HubName)
+        Throw ("FinOps Hub {0} Not Found" -f $HubName)
+    }
+
+    Write-Output ("{0}    FinOps Hub {1} Found" -f (Get-Date), $HubName)
+
+    $storageAccount = Get-AzStorageAccount -ResourceGroupName $HubName -ErrorAction SilentlyContinue
+    if ($null -eq $storageAccount) {
+        Write-Output ("{0}    Storage Account Not Found" -f (Get-Date))
+        Throw ("Storage Account Not Found")
+    }
+
+    if ($storageAccount.Count -gt 1) {
+        Write-Output ("{0}    Multiple Storage Accounts Found" -f (Get-Date))
+        Throw ("Multiple Storage Accounts Found")
+    } # handle this better later on to select the correct one.
+
+    Write-Output ("{0}    Storage Account Found" -f (Get-Date))
+
+    $storageContext = $StorageAccount.Context
+    Get-AzStorageBlob -Container 'config' -Blob 'settings.json' -Context $storageContext | Get-AzStorageBlobContent -Force | Out-Null
+    $settings = Get-Content 'settings.json' | ConvertFrom-Json
+
+    if (($settings.exportScopes.Count -eq 1) -and ([string]::IsNullOrEmpty($settings.exportScopes[0]))) {
+        $settings.exportScopes = @()
+        $operation = 'create'
+    }
+
+    foreach ($ScopeToUpdate in $settings.exportScopes) {
+        if ($ScopeToUpdate.scope -eq $Scope) {
+            Write-Output ("{0}    Export Scope {1} Already Exists" -f (Get-Date), $Scope)
+            if ($ScopeToUpdate.tenantId -eq $TenantId) {
+                Write-Output ("{0}    tenantId {1} Matches" -f (Get-Date), $TenantId)
+                $operation = 'none'
+            } else {
+                Write-Output ("{0}    tenantId {1} <> {2}" -f (Get-Date), $ScopeToUpdate.tenantId, $TenantId)
+                $operation = 'update'
+            }
+
+            if ($ScopeToUpdate.cloud -eq $Cloud) {
+                Write-Output ("{0}    cloud {1} Matches" -f (Get-Date), $Cloud)
+                $operation = 'none'
+            } else {
+                Write-Output ("{0}    cloud {1} <> {2}" -f (Get-Date), $ScopeToUpdate.cloud, $Cloud)
+                $operation = 'update'
+            }
+        } else {
+            Write-Output ("{0}    Export Scope {1} Skipped" -f (Get-Date), $ScopeToUpdate.scope)
```

Comment:
Should this explain what it means to be skipped? That might be perceived as a bug.
```suggestion
            Write-Output ("{0}    Export scope {1} skipped" -f (Get-Date), $ScopeToUpdate.scope)
```